### PR TITLE
Alternative parametrization of GaussianSquare

### PR DESCRIPTION
**Pull request by Feigenbaum4669**
### Summary
A previous version of GaussianSquare pulse used four parameters to determine the signal: amplitude, sigma, duration, width.  In the new version there is additional parameter - risefall_sigma_ratio (the duration of risefall divided by sigma). Now the user has to specify either width ("fixed width mode") or risefall_sigma_ratio parameter ("fixed risefall mode"; assuming sigma-fixed), both of which are set by default to None. This way one can choose between old (fixed width) and new, more natural, parametrization ("fixed risefall mode") where the total length of a risefall is fixed wrt to changes in duration.  As a consequence, in fixed risefall mode an increase in duration is realized by an increase in width (i.e. constant part of the signal).  In particular,  the integral of the signal grows linearly with its duration. The initial behaviour ("fixed width mode") realized an increase in duration by an increase in risefall duration.

### Details and comments

- If both width and risefall_sigma_ratio are not None a PulseError is raised. 
- Similarly, if both are None. 
- If width is None then its value is computed as width = duration - 2 * risefall_sigma_ratio * sigma.
-  If risefall_sigma_ratio is None then its value is computed as risefall_sigma_ratio=(duration-width)/(2*sigma). 
- The "parameters" field and __repr__(self) have not been changed (i.e. width is preserved; risefall_to_sigma is not present ). 
- The parameter risefall_sigma_ratio can take values > 0 and < duration/(2*sigma), which is checked in  "validate_parameters". 
- Please find attached Figures showing the difference between the two modes when the duration is doubled. The width and risefall_sigma_ratio parameters are chosen such that they correspond to the same signal for a shorter duration (600 dt).
-  I have added some tests to cover the new functionality.

w500d600
w500d1200
rts2 5w500d600
rts2 5w1100d1200

### CODE REVIEW

**Comment by taalexander** [comment]
Fixes #6395.

File: qiskit/pulse/library/parametric_pulses.py [hunk 1]
```diff
@@ -208,14 +221,16 @@ class GaussianSquare(ParametricPulse):
...
     """
 
     def __init__(
         self,
         duration: Union[int, ParameterExpression],
         amp: Union[complex, ParameterExpression],
         sigma: Union[float, ParameterExpression],
-        width: Union[float, ParameterExpression],
+        width: Union[float, ParameterExpression] = None,
+        risefall_to_sigma: Union[float, ParameterExpression] = None,
```
Comment (taalexander) [inline qiskit/pulse/library/parametric_pulses.py:233]: I wonder if `risefall_over_sigma` or `risefall_sigma_ratio` are clearer?

File: qiskit/pulse/library/parametric_pulses.py [hunk 2]
```diff
@@ -256,12 +278,42 @@ def get_waveform(self) -> Waveform:
...
+
         if not _is_parameterized(self.amp) and abs(self.amp) > 1.0:
             raise PulseError("The amplitude norm must be <= 1, " "found: {}".format(abs(self.amp)))
         if not _is_parameterized(self.sigma) and self.sigma <= 0:
             raise PulseError("Sigma must be greater than 0.")
-        if not _is_parameterized(self.width) and (self.width < 0 or self.width >= self.duration):
-            raise PulseError("The pulse width must be at least 0 and less than its duration.")
+        if self.width is not None and self.risefall_to_sigma is not None:
+            raise PulseError(
+                "Either the pulse width or the risefall_to_sigma parameter can be specified."
```
Comment (taalexander) [inline qiskit/pulse/library/parametric_pulses.py:288]:
```suggestion
                "Either the pulse width or the risefall_to_sigma parameter can be specified but not both."
```

File: qiskit/pulse/library/parametric_pulses.py [hunk 3]
```diff
@@ -256,12 +278,42 @@ def get_waveform(self) -> Waveform:
...
+            )
+        if self.width is None and self.risefall_to_sigma is None:
+            raise PulseError(
+                "Either the pulse width or the risefall_to_sigma parameter has to be specified."
```
Comment (taalexander) [inline qiskit/pulse/library/parametric_pulses.py:292]:
```suggestion
                "Either the pulse width or the risefall_to_sigma parameter must be specified."
```

**Review by taalexander — CHANGES_REQUESTED** [review]
Looks, good. Thoughts on the parameter name @oliverdial?

File: qiskit/pulse/library/parametric_pulses.py (hunk 1 above)
Comment (Feigenbaum4669) [reply qiskit/pulse/library/parametric_pulses.py:233]: perhaps risefall_to_sigma_ratio?

Comment (taalexander) [reply qiskit/pulse/library/parametric_pulses.py]: On second thought `risefall_to_sigma` is just as good as other suggestions 😄

**Review by taalexander — APPROVED** [review]

**Pull request merged**